README.adoc

```diff
@@ -0,0 +1,187 @@
+tBTC:
++
+....
+@keep-network:registry=https://npm.pkg.github.com/keep-network
+....
+3.  Install the package:
++
+[source,sh]
+----
+$ npm install @keep-network/tbtc
+----
+
+== Usage
+
+*NOTE:* tBTC contracts require _solc_ v0.5.10 or higher. You may have to
+https://www.trufflesuite.com/docs/truffle/reference/configuration#compiler-configuration[configure
+solc in your `truffle-config.js`].
+
+Once installed, you can use the contracts in the library by importing
+them:
+
+[source,sol]
+----
+pragma solidity ^0.5.0;
+
+import "@keep-network/tbtc/contracts/deposit/Deposit.sol";
+
+contract MySystem {
+    function checkTerm(address _depositAddress) external {
+        uint256 remainingTerm = Deposit(_depositAddress).remainingTerm();
+    }
+}
+----
+
+== Security
+
+tBTC is currently under audit.
+
+Please report any security issues you find to mailto:security@keep.network[security@keep.network].
+
+== Contributing
+
+All contributions are welcome. To report bugs, please create an issue on
+this repository. To start a discussion, prefer Discord over GitHub
+issues.
+
+*Read the
+https://github.com/keep-network/tbtc/blob/master/CONTRIBUTING.md[Contributing
+guidelines].*
+
+=== Setup environment
+
+You should have installed:
+
+* Node.js, https://docs.npmjs.com/cli/install[npm].
+* A local Ethereum blockchain. We recommend
+https://www.trufflesuite.com/ganache[Ganache].
+* https://www.trufflesuite.com/docs/truffle/overview[Truffle framework].
+
+=== Build
+
+Clone and install dependencies:
+
+[source,sh]
+----
+git clone https://github.com/keep-network/tbtc
+cd tbtc/implementation
+npm install
+----
+
+Deploy contracts:
+
+[source,sh]
+----
+truffle migrate --reset
+----
+
+=== Test
+
+Tests are written in JS using Mocha.
+
+To run the test suite, execute `truffle test`.
+
+To run specific tests, add
+https://jaketrent.com/post/run-single-mocha-test/[`.only`] to the
+`contract` block:
+
+[source,js]
+----
+contract.only('TBTCToken', function(accounts) {
+----
+
+=== Lint
+
+We use https://eslint.org/[ESLint] and
+https://github.com/duaraghav8/Ethlint[Ethlint] for linting code. To run:
+
+[source,sh]
+----
+npm run sol:lint:fix
+npm run js:lint:fix
+----
+
+== Documentation
+
+The documentation includes a project overview and rationale, as well as
+the on-chain specification. Docs should always be updated before or in
+tandem with code.
+
+=== Prerequisites
+
+Docs are written in http://asciidoctor.org/[AsciiDoctor], diagrams in a
+LaTeX package called
+https://www.overleaf.com/learn/latex/TikZ_package[Tikz].
+
+==== macOS
+
+1.  Install https://www.tug.org/texlive/[TeX Live] manually, and other
+dependencies using CLI:
++
+[source,sh]
+----
+gem install asciidoctor-pdf --pre
+brew install poppler
+----
+2.  Install the TikZ package to your local LaTeX environment:
++
+[source,sh]
+----
+sudo cp docs/latex/tikz-uml.sty /usr/local/texlive/texmf-local/
+
+# Update TeX package tree
+sudo texhash
+----
+
+=== Build
+
+[source,sh]
```

Comment:
Worth noting, asciidoc supports (as far as I remember) the markdown-style triple-backtick-followed-by-syntax-type code fences as well. No need to change this, necessarily, just flagging it.